orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/alicloud/AliCloudServerGroupCreator.groovy

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2019 Alibaba Group.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.clouddriver.tasks.providers.alicloud
+
+import com.netflix.spinnaker.orca.clouddriver.tasks.servergroup.ServerGroupCreator
+import com.netflix.spinnaker.orca.kato.tasks.DeploymentDetailsAware
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import org.springframework.stereotype.Component
+
+@Component
+class AliCloudServerGroupCreator implements ServerGroupCreator, DeploymentDetailsAware {
```

Comment:
Please do not introduce new Groovy files into the `main` source set. Use Java instead.

Reply:
HI @robertsigler I also want to use Java, but I found `DeploymentDetailsAware` is Groovy owned type and I can not implement it in the Java. Do you have any idea?

Reply:
That shouldn't be the case. We implement Groovy interfaces with Java all over the codebase.

Reply:
Hi @robzienert , DeploymentDetailsAware is not a common Java interface but a Groovy trait, if I implement it , I must rewrite the default implement in DeploymentDetailsAware. Here is the usage of trait from Groovy docs:
"This means that even if a trait can be seen from a Java class as a regular interface, that interface will not have default methods, only abstract ones."